### PR TITLE
Swift Code Generation

### PR DESCRIPTION
**Pull request by JoshHrach**
This PR adds support for viewing Swift code to reproduce the `MeshGradient`.

It also adds support for deleting gradient colors from the inspector.

### CODE REVIEW

File: MeshBuddy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -373,7 +385,7 @@
 			buildSettings = {
 				CODE_SIGN_ENTITLEMENTS = MeshQuickLook/MeshQuickLook.entitlements;
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = 8C7439RJLG;
+				DEVELOPMENT_TEAM = QQK58F3XAD;
```
Comment (JoshHrach) [inline MeshBuddy.xcodeproj/project.pbxproj]: Oops, forgot this change. I'll revert.

**Comment by insidegui** [comment]
Thanks for building this, @JoshHrach!

I've noticed a couple of things I'd like you to check out before we merge this.

### File Menu

Because of the new window type, there's now a "New Window" item under File > New that just creates an empty window. I think this can be addressed by using a `Window` instead of a `WindowGroup` for the Swift code window.

MeshBuddy 2024-12-03 09 47 43

### Color Mismatch

The generated code results in a mesh gradient with washed out colors when compared to the original. At first I thought this was because of the `colorSpace` option, but changing it between `.device` and `.perceptual` didn't make a difference.

My guess is this is due to the initializer used for `Color` in the generated code. You'll probably want to use an initializer that interprets the color values as P3.

I think you'll have to be on a P3 display in order to be able to see the difference:

**Reference:**

reference

**Generated Code:**

generated

**Comment by JoshHrach** [comment]
Thanks for reviewing. I expected feedback. 😅

I typically use my LG 4K for my dev work, and I don't think it supports P3. I'll use my MacBook's screen to confirm. I think I have an idea as to what you're seeing, though, and I'll work on addressing that.

Thanks for catching the Window, too.

**Comment by JoshHrach** [comment]
The default color space for `Color` is `.sRGB`. Instead of going to the `NSColor`, I'm now resolving the color and using the provided sRGB values. Please test on your P3 and see if this improves things. On my (M1 Max) MacBook Pro's screen, it looks better.

**Comment by insidegui** [comment]
Looks great now! Thanks 🙌🏻

**Pull request merged**